packages/js/src/dashboard/routes/first-time-configuration.js

```diff
@@ -3,24 +3,28 @@ import { Paper } from "@yoast/ui-library";
 import FirstTimeConfigurationSteps from "../../first-time-configuration/first-time-configuration-steps";
 import {
 	RouteLayout,
+	UnsavedChangesModal,
 } from "../components";
 
 /**
  * @returns {JSX.Element} The site defaults route.
  */
 const FirstTimeConfiguration = () => {
 	return (
-		<Paper>
-			<RouteLayout
-				title={ __( "First-time configuration", "wordpress-seo" ) }
-				description={ __( "Tell us about your site, so we can get it ranked! Let's get your site in tip-top shape for the search engines. Follow these 5 steps to make Google understand what your site is about.", "wordpress-seo" ) }
-			>
-				<hr id="configuration-hr-top" />
-				<div id="yoast-configuration" className="yst-p-8 yst-max-w-[715px]">
-					<FirstTimeConfigurationSteps />
-				</div>
-			</RouteLayout>
-		</Paper>
+		<>
+			<Paper>
+				<RouteLayout
+					title={ __( "First-time configuration", "wordpress-seo" ) }
+					description={ __( "Tell us about your site, so we can get it ranked! Let's get your site in tip-top shape for the search engines. Follow these 5 steps to make Google understand what your site is about.", "wordpress-seo" ) }
+				>
+					<hr id="configuration-hr-top" />
+					<div id="yoast-configuration" className="yst-p-8 yst-max-w-[715px]">
+						<FirstTimeConfigurationSteps />
+					</div>
+				</RouteLayout>
+			</Paper>
+			<UnsavedChangesModal />
```

Comment:
Just a side note: this does not necessarily _need_ to be outside of the Paper, as it renders a portal somewhere else anyway.